test/powershell/Modules/Microsoft.PowerShell.Utility/JsonObject.Tests.ps1

```diff
@@ -1,46 +1,31 @@
 ﻿Describe 'Unit tests for JsonObject' -tags "CI" {
 
-    function ShouldThrow
-    {
-        param (
-            [Parameter(ValueFromPipeline = $true)]
-            $InputObject,
-            [Parameter(Position = 0)]
-            $ExpectedException
-        )
-
-        try
-        {
-            & $InputObject
-            throw "Should throw exception"
-        }
-        catch
-        {
-            $_.FullyQualifiedErrorId | should be $ExpectedException
-        }
-    }
-
     $validStrings = @(
-        @{ name = "empty"; str = "" }
-        @{ name = "spaces"; str = "  " }
-        @{ name = "object"; str = "{a:1}" }
+        @{ name = "empty";  str = "";      ReturnHashTable = $true  }
+        @{ name = "spaces"; str = "  ";    ReturnHashTable = $true  }
+        @{ name = "object"; str = "{a:1}"; ReturnHashTable = $true  }
+        @{ name = "empty";  str = "";      ReturnHashTable = $false }
+        @{ name = "spaces"; str = "  ";    ReturnHashTable = $false }
+        @{ name = "object"; str = "{a:1}"; ReturnHashTable = $false }
     )
 
-    It 'no error for valid string - <name>' -TestCase $validStrings {
-        param ($str)
+    It 'no error for valid string ''<name>'' with -ReturnHashTable:$<ReturnHashTable>' -TestCase $validStrings {
+        param ($str, $ReturnHashTable)
         $errRecord = $null
-        [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, [ref]$errRecord)
+        [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, $ReturnHashTable, [ref]$errRecord)
         $errRecord | Should BeNullOrEmpty
     }
 
     $invalidStrings = @(
-        @{ name = "plain text"; str = "plaintext" }
-        @{ name = "part"; str = '{"a" :' }
+        @{ name = "plain text"; str = "plaintext"; ReturnHashTable = $true  }
+        @{ name = "part";       str = '{"a" :';    ReturnHashTable = $true  }
+        @{ name = "plain text"; str = "plaintext"; ReturnHashTable = $false }
+        @{ name = "part";       str = '{"a" :';    ReturnHashTable = $false }
```

Comment:
I see we have a check for `input == null` but haven't the test - please add new test with `str = $null`.

Reply:
OK. Added